src/main/java/com/cityelf/service/WaterForecastService.java

```diff
@@ -0,0 +1,91 @@
+package com.cityelf.service;
+
+import com.cityelf.exceptions.ForecastAlreadyExistsException;
+import com.cityelf.exceptions.ForecastNotFoundException;
+import com.cityelf.model.Address;
+import com.cityelf.model.WaterForecast;
+import com.cityelf.repository.WaterForecastRepository;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Service
+public class WaterForecastService {
+
+  @Autowired
+  private WaterForecastRepository waterForecastRepository;
+
+  public WaterForecastService() {
+  }
+
+  //Get all water forecasts in repository. If we clear repository every few hours, it will return only
+  //actual or just ended forecasts.
+  public Iterable<WaterForecast> getAll() {
+    return waterForecastRepository.findAll();
+  }
+
+  //Get water forecast by ID
+  public WaterForecast getForecast(long id) throws ForecastNotFoundException {
+    WaterForecast forecast = waterForecastRepository.findOne(id);
+    if (forecast == null) {
+      throw new ForecastNotFoundException();
+    } else {
```

Comment:
You don't need **else** the same as in the other similar constructors. It is not a mistake, just a redundancy.